File: week-04/day-01/sharpie.py
class Sharpie(object):
    def __init__(self, color, width):
        self.color = color
        self.width = width
        self.ink_amount = 100

    def use(self):
        self.ink_amount -= 1

    def __str__(self):
        return '[Color: {}, Width: {}, Ink amount: {}]'.format(self.color, self.width, self.ink_amount)


red = Sharpie("Red", 20)
print(red)
for i in range(73):
    red.use()
print(red)

File: week-09/day-03/sharpie.js
'use strict'

class Sharpie {

  constructor(color, width) {
    this.color = color;
    this.width = width;
    this.inkAmount = 100;
  }

  use() {
    this.inkAmount -= this.width;
  }
}

let red = new Sharpie('red', 11);
let yellow = new Sharpie('yellow', 32);

function useSharpie(color, inkAmount, width) {
  if (typeof inkAmount !== 'number' || typeof width !== 'number') {
    throw new Error('error');
  }
  for (let i = inkAmount; i >= width; i -= width) {
    color.inkAmount -= color.width;
  }
  return `${color.color} ink amount left: ${color.inkAmount}`;
}

console.log(useSharpie(red, red.inkAmount, red.width));
console.log(useSharpie(yellow, yellow.inkAmount, yellow.width));

